File: competitive_programming/leetcode/06_javascript/242_valid_anagram.js
function validAnagram(s, t) {
  return s.split("").sort().join("") === t.split("").sort().join("");
}

if (require.main === module) {
  if (validAnagram("anagram", "nagaram") !== true || validAnagram("rat", "car") !== false) throw new Error("valid anagram failed");
  console.log("242 valid anagram ok");
}

File: competitive_programming/leetcode/09_rust/242_valid_anagram.rs
fn valid_anagram(s: &str, t: &str) -> bool {
    let mut left: Vec<char> = s.chars().collect();
    let mut right: Vec<char> = t.chars().collect();
    left.sort();
    right.sort();
    left == right
}

fn main() {
    assert!(valid_anagram("anagram", "nagaram"));
    assert!(!valid_anagram("rat", "car"));
    println!("242 valid anagram ok");
}
